test/lib/foreman/logging_test.rb

```diff
@@ -7,6 +7,23 @@ def test_configure_once
     end
   end
 
+  def test_environment_not_found_yields_defaults
+    assert_nothing_raised do
+      Foreman::Logging.instance_variable_set("@configured", false)
+      # Stub the log file creation (foo.log)
+      Foreman::Logging.stubs(:build_file_appender).
+        with('foreman', { :environment => 'foo'} ).returns(nil).once
+      quietly { Foreman::Logging.configure(:environment => 'foo') }
+    end
+    log_filename = Foreman::Logging.instance_variable_get("@config")[:filename]
+    assert_equal 'foo.log', log_filename
+  ensure
+    # Return the foreman logger to the original one for the current environment
```

Comment:
Sorry, this is a bit extreme I think as it changes too much of the application's global state just to run a test.  Either see if you can make the class more testable or remove it, but I think this is a bit too dangerous to leave in the test suite.